central/xlsx/reader.go

```diff
@@ -209,8 +165,40 @@ func convertDeviceFields(rowElement *reflect.Value, xlsxRow []string, headerCol
 
 					switch fieldPrefix {
 					case strings.ToLower(protocols):
-						// set the cell to Protocols map
-						protocolProperties[headerName] = convertedValue
+						// get the Device protocols field
+						prtMapField := rowElement.FieldByName(protocols)
+						if prtMapField.Len() > 0 {
+							// convert the prtMapField reflect.Value to map[string]ProtocolProperties
+							prtPropMap = prtMapField.Interface().(map[string]dtos.ProtocolProperties)
```

Comment:
```suggestion
							prtPropMap, ok = prtMapField.Interface().(map[string]dtos.ProtocolProperties)
							if !ok {
								return error
							}
```

check any conversion to prevent panic